.github/workflows/azure-e2e-test.yml

```diff
@@ -5,8 +5,8 @@ permissions:
   contents: read
 
 env:
-  CLUSTER_NAME: "e2e-test-${{ github.run_id }}-${{ github.run_attempt }}"
-  TEST_PROVISION_FILE: "${{ github.workspace }}/provision_azure${{ github.run_id }}.properties"
+  CLUSTER_NAME_SUFFIX: "e2e-test-${{ github.run_id }}-${{ github.run_attempt }}"
+  TEST_PROVISION_FILE_PREFIX: "${{ github.workspace }}/provision_azure${{ github.run_id }}.properties"
```

Comment:
Nit: Maybe this can be converted into `TEST_PROVISION_PATH_TEMPLATE`, to keep the. properties extension.

```
TEST_PROVISION_PATH_TEMPLATE: "${{ github.workspace }}/provision_azure_${{ github.run_id }}_{0}.properties"
```

then we can use 

```
${{ env[format(env.TEST_PROVISION_PATH_TEMPLATE, matrix.parameters.id)] }}
```

to reference it.

Reply:
similarly for the cluster name as well!